zerver/decorator.py

```diff
@@ -717,15 +717,20 @@ def _wrapped_view_func(request: HttpRequest, *args: object, **kwargs: object) ->
         if not request.POST:
             # Only take action if POST is empty.
             if request.content_type == "multipart/form-data":
-                # Note that request._files is just the private attribute that backs the
-                # FILES property, so we are essentially setting request.FILES here.  (In
-                # Django 1.5 FILES was still a read-only property.)
-                request.POST, request._files = MultiPartParser(
+                POST, _files = MultiPartParser(
                     request.META,
                     BytesIO(request.body),
                     request.upload_handlers,
                     request.encoding,
                 ).parse()
+                # request.POST is immutable in most cases. We need to bypass this restriction
+                # since we are initializing it on our own.
+                # See also: https://github.com/typeddjango/django-stubs/pull/925#issue-1206399444
+                setattr(request.POST, "POST", POST)
```

Comment:
Also, this comment and link don’t make sense as an explanation. We are mutating the `HttpRequest` (`request.POST = …`); the immutability of `QueryDict` (`request.POST["key"] = …`) is irrelevant.

In fact, I don’t see any reason why mypy shouldn’t accept `request.POST = POST`.

Reply:
@andersk thoughts on this version? I could imagine a mypy ignore, or a maybe a cast from the mutable type to the immutable type, being a cleaner solution.

Reply:
I tend to agree. If we are manually changing `POST` mutability, we should also manually `cast` to acknowledge this change. `_files` could be added to django-stubs upstream.

We could skip the `decorator.py` change for now and merge the rest?

Here’s my question about this code:

https://github.com/zulip/zulip/blob/a142fbff85302c5e3acb2e204eca2e9c75dbc74b/zerver/decorator.py#L714-L715

Was that bug ever filed?

Reply:
@andersk This was reported [here](https://code.djangoproject.com/ticket/15262) and [here](https://code.djangoproject.com/ticket/12635). There is no plan to fix this as far as I am concerned.

Regarding the mutability, there is an [issue](https://code.djangoproject.com/ticket/17235) upstream and a [PR](https://github.com/django/django/pull/10110) fixing it. But no progress has been made since 2018.

I updated the PR with `cast`. There can be a follow-up to this if both of the issues are resolved.